### PR TITLE
BUG: repr SparseDataFrame after setting a value

### LINKED ISSUE #15488: Repr-ing SparseDataFrame after setting a value
#### Code Sample, a copy-pastable example if possible

```python
>>> sdf = pd.SparseDataFrame(index=[0, 1], columns=[0, 1])
>>> sdf[0][0] = 1  # This warns; ignore
>>> sdf
AttributeError: 'NoneType' object has no attribute 'type'
```
#### Problem description

After some value is set on a copy of a SparseSeries, the enclosing SparseDataFrame can no longer be printed (raises AttributeError).

#### Expected Output
```py
>>> sdf
    0   1
0 NaN NaN
1 NaN NaN
```
#### Output of ``pd.show_versions()``

0.19.0+479.git

### PR DESCRIPTION
**Pull request by kernc**
- [x] closes #15488
 - [x] tests added / passed
 - [X] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/sparse/test_format.py [hunk 1]
```diff
@@ -116,3 +116,16 @@ def test_sparse_frame(self):
 
         with option_context("display.max_rows", 3):
             self.assertEqual(repr(sparse), repr(df))
+
+    def test_sparse_repr_after_set(self):
+        # GH 15488
+        sdf = pd.SparseDataFrame(index=[0, 1], columns=[0, 1])
+
```
Comment (jreback) [inline pandas/tests/sparse/test_format.py]: use ``pd.option_context`` as a context manager instead

File: pandas/tests/sparse/test_format.py [hunk 2]
```diff
@@ -116,3 +116,16 @@ def test_sparse_frame(self):
...
+        # Ignore the warning
+        old_opt = pd.core.config.get_option('mode.chained_assignment')
+        pd.core.config.set_option('mode.chained_assignment', None)
+        sdf[0][0] = 1  # This line triggers the bug
+        try:
+            repr(sdf)
+        finally:
+            pd.core.config.set_option('mode.chained_assignment', old_opt)
```
Comment (jreback) [inline pandas/tests/sparse/test_format.py]: also can you use ``assert_sp_frame_equal`` and compare

File: pandas/tests/sparse/test_format.py (hunk 1 above)
Comment (kernc) [inline pandas/tests/sparse/test_format.py]: Thanks! Updated.

**Comment by jreback** [comment]
ok, lgtm. ping on green.

**Comment by kernc** [comment]
It's green.

**Comment by jreback** [comment]
thanks @kernc 

keep em coming!

**Pull request closed without merging**